lib/http_accept_language/auto_locale.rb

```diff
@@ -11,7 +11,9 @@ module AutoLocale
     private
 
     def set_locale
-      I18n.locale = http_accept_language.compatible_language_from(I18n.available_locales)
+      hal = http_accept_language
+      hal.header ||= I18n.default_locale.to_s
+      I18n.locale = hal.compatible_language_from(I18n.available_locales)
```

Comment:
In this case, I would assume you would want to still present your website to the user using the default locale (usually English) rather than error out.